src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -248,17 +263,44 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @return true if a right curly should be alone on a line.
      */
     private static boolean shouldBeAloneOnLine(RightCurlyOption bracePolicy, Details details) {
+        final int tokenLambda = details.lcurly.getParent().getType();
         return bracePolicy == RightCurlyOption.ALONE
+                && tokenLambda != TokenTypes.LAMBDA
                 && !isAloneOnLine(details)
                 && !isEmptyBody(details.lcurly)
-                || bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
+                || shouldBeAloneInLambda(bracePolicy, details)
+                || shouldBeAloneOrSingleLine(bracePolicy, details);
+    }
```

Comment:
@romani To skip the check "cyclomatic complexity" in all methods, I propose to create a method common to all  methods with the word "Alone" in names. 
 _private static boolean shouldBeAloneOnLineGeneral(RightCurlyOption bracePolicy,
                                                      Details details) {
        final int tokenLambda = details.lcurly.getParent().getType();
        return tokenLambda != TokenTypes.LAMBDA
                && shouldBeAloneOnLine(bracePolicy, details)
                || tokenLambda == TokenTypes.LAMBDA
                && shouldBeAloneOther(bracePolicy, details);
    }_

And in **validate else if (shouldBeAloneOnLineGeneral(bracePolicy, details)) {
            violation = MSG_KEY_LINE_ALONE;
        }**


this option acceptable?